crunchy.go

```diff
@@ -142,12 +143,27 @@ func (v *Validator) foundInDictionaries(s string) error {
 	// find mangled / reversed passwords
 	// we can skip this if the pw is longer than the longest word plus our minimum distance
 	if pwlen <= v.wordsMaxLen+v.options.MinDist {
+		jobs := runtime.GOMAXPROCS(0)
+		queue := make(chan struct{}, jobs)
+		errc := make(chan *DictionaryError, jobs)
 		for word := range v.words {
-			if dist := smetrics.WagnerFischer(word, pw, 1, 1, 1); dist <= v.options.MinDist {
-				return &DictionaryError{ErrMangledDictionary, word, dist}
-			}
-			if dist := smetrics.WagnerFischer(word, revpw, 1, 1, 1); dist <= v.options.MinDist {
-				return &DictionaryError{ErrMangledDictionary, word, dist}
+			select {
+			case queue <- struct{}{}:
+				go func(word string) {
+					var err *DictionaryError
+					if dist := smetrics.WagnerFischer(word, pw, 1, 1, 1); dist <= v.options.MinDist {
+						err = &DictionaryError{ErrMangledDictionary, word, dist}
```

Comment:
We never check the error returned here. We can already abort the check here if an error occurred. Also, why don't we simply write to `errc` here directly?

Reply:
The intention was to avoid sending more than one value over the channel per run, because that could potentially leak goroutines as the `errc` channel has a buffer of `njobs`.

I think I can make this part less ambiguous with your suggestion. Will push a fix.